src/header.rs

```diff
@@ -206,6 +214,18 @@ impl TbfHeader {
                 tipe: TbfHeaderTypes::PackageName,
                 length: self.package_name.len() as u16,
             });
+            hasher.update(self.package_name.clone());
+        }
+
+        // Generate an AppID from the package name
+        if app_id.is_none() {
+            let hash = hasher.finalize();
+            self.hdr_main.app_id = hash[0] as u32
+                | (hash[1] as u32) << 8
+                | (hash[2] as u32) << 16
+                | (hash[3] as u32) << 24;
+        } else {
+            self.hdr_main.app_id = app_id.unwrap();
```

Comment:
If the package name does not exist this will use the default value inside the hasher, which could lead to two unnamed packages having the same AppId. The same problem would exist for two packages with the same name being flashed. This seems like it could be a pretty big problem if the kernel assumed that passed appids do not collide.

Reply:
Hmm... Good point.

The goal was so that casual users would generally end up with app-id's that are constant over updates and layout changes.

I could swap this to just be a random number, but then apps will never have the same ID unless the user manually specifies it. I'm not sure if that matters much though.

I'll look into this more

Reply:
> The goal was so that casual users would generally end up with app-id's that are constant over updates and layout changes.

I think that this is a large part of why we want to have application ids in the first place. I'd refrain from using something that cannot be user controlled. See [my comment](https://github.com/tock/elf2tab/pull/27#issuecomment-715112111) for a further explanation of my reasoning.